### PR TITLE
feat: Improve checkpointing system

### PR DESCRIPTION
**Pull request by Wolmin**
https://app.clickup.com/t/2645698/DEV-5553

### CODE REVIEW

**Review by richtera — APPROVED** [review]
LGTM

File: commands/start.go [hunk 1]
```diff
@@ -172,3 +196,54 @@ func startValidator(ctx *cli.Context) (err error) {
...
+	err = json.Unmarshal(resByte, &rootResp)
+	if err != nil {
+		return
+	}
+
+	var finalizedSlotInt int
+	finalizedSlotInt, err = strconv.Atoi(finalizedSlot)
+
+	finalizedRoot = "0x" + rootResp.Data.Root
+	epoch = finalizedSlotInt / 32
```
Comment (mxmar) [inline commands/start.go]: Please move this `32` to consts and call it like: `slotsPerEpoch`

**Review by mxmar — APPROVED** [review]

**Review by mxmar — CHANGES_REQUESTED** [review]
@Wolmin please update `README.md` about weak subjectivity feature.

File: commands/start.go (hunk 1 above)
Comment (Wolmin) [reply commands/start.go]: Done - also wrote it in the `common` package

**Review by mxmar — APPROVED** [review]
LGTM now :+1:

**Comment by fhildeb** [comment]
Please rename your feat commit to fix- so release automation is working as we want. Otherwise, looks good

**Review by fhildeb — APPROVED** [review]

**Pull request merged**